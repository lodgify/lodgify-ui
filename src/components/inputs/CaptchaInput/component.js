import React from 'react';
import PropTypes from 'prop-types';
import { Form, Image } from 'semantic-ui-react';

import { TextInput } from 'inputs/TextInput';

/**
 * A captcha input helps to ensure a user is a real human.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  value,
  error,
  image,
  isValid,
  label,
  name,
  onBlur,
  onChange,
}) => (
  <Form.Group widths="equal">
    <Form.Field>
      <Image src={image} />
    </Form.Field>
    <Form.Field>
      <TextInput
        autoComplete="off"
        error={error}
        isValid={isValid}
        label={label}
        name={name}
        onBlur={onBlur}
        onChange={onChange}
        value={value}
      />
    </Form.Field>
  </Form.Group>
);

Component.displayName = 'CaptchaInput';

Component.defaultProps = {
  error: false,
  isValid: false,
  label: '',
  name: '',
  onBlur: Function.prototype,
  onChange: Function.prototype,
  value: undefined,
};

Component.propTypes = {
  /** Is the captcha input in an error state. */
  error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  /** The source url for the image to display. */
  image: PropTypes.string.isRequired,
  /** Is the captcha input in a valid state. */
  isValid: PropTypes.bool,
  /** The visible label for the text input. */
  label: PropTypes.string,
  /** The name for the captcha input. */
  name: PropTypes.string,
  /**
   * Used internally by `Form` so ignored in the styleguide.
   * @ignore
   */
  onBlur: PropTypes.func,
  /**
   * A function called when the captcha input value changes
   * @param {String} name
   * @param {String} value
   */
  onChange: PropTypes.func,
  /** The current value of the input where it is consumed as a controlled component. */
  value: PropTypes.oneOfType([
    PropTypes.bool,
    PropTypes.string,
    PropTypes.number,
  ]),
};
