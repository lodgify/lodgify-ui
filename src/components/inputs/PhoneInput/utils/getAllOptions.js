import React, { Fragment } from 'react';

import { getIconOrFlag } from './getIconOrFlag';
import { getDiallingCode } from './getDiallingCode';

const optionCache = {};

/**
 * @param  {Object[]} options
 * @return {Object[]}
 */
export const getAllOptions = options =>
  options.map(({ value, label }) => {
    if (!optionCache[label]) {
      const iconOrFlag = getIconOrFlag(value);

      optionCache[label] = {
        name: label,
        text: iconOrFlag,
        value,
        content: (
          <Fragment>
            {iconOrFlag}
            {`${label} ${getDiallingCode(value)}`}
          </Fragment>
        ),
      };
    }

    return optionCache[label];
  });
