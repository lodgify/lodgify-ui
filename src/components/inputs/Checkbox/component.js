import React from 'react';
import PropTypes from 'prop-types';
import { Checkbox } from 'semantic-ui-react';

import { InputController } from '../InputController';

import { mapValueToProps } from './utils/mapValueToProps';
import { adaptOnChangeEvent } from './utils/adaptOnChangeEvent';

/**
 * A checkbox allows a user to select a value from a small set of options, often binary.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  error,
  isDisabled,
  isChecked,
  isRadioButton,
  isToggle,
  isValid,
  label,
  name,
  onChange,
}) => (
  <InputController
    adaptOnChangeEvent={adaptOnChangeEvent}
    error={error}
    isValid={isValid}
    label={label}
    mapValueToProps={mapValueToProps}
    name={name}
    onChange={onChange}
    value={isChecked}
  >
    <Checkbox disabled={isDisabled} radio={isRadioButton} toggle={isToggle} />
  </InputController>
);

Component.displayName = 'Checkbox';

Component.defaultProps = {
  error: false,
  isChecked: undefined,
  isDisabled: false,
  isToggle: false,
  isRadioButton: false,
  isValid: false,
  label: '',
  name: '',
  onChange: Function.prototype,
};

Component.propTypes = {
  /** Is the checkbox input in an error state. */
  error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  /** Is the checkbox checked. */
  isChecked: PropTypes.bool,
  /** Is the checkbox disabled. */
  isDisabled: PropTypes.bool,
  /** Is the checkbox formatted as a radio button. Prefer using the <RadioButton> component instead */
  isRadioButton: PropTypes.bool,
  /** Format to show an on or off choice. */
  isToggle: PropTypes.bool,
  /** Is the checkbox in a valid state. */
  isValid: PropTypes.bool,
  /** The label for the checkbox */
  label: PropTypes.node,
  /** The HTML input name. */
  name: PropTypes.string,
  /**
   * Event called when the checkbox value changes.
   * @param {String} name
   * @param{boolean} value
   */
  onChange: PropTypes.func,
};
