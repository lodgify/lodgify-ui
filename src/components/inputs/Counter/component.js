import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { FlexContainer } from 'layout/FlexContainer';
import { Button } from 'elements/Button';
import { ICON_NAMES } from 'elements/Icon';
import { Paragraph } from 'typography/Paragraph';
import { getControlledInputValue } from 'utils/get-controlled-input-value';

/**
 * A counter allows a user to select a number value in increments of one.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export class Component extends PureComponent {
  state = {
    value: this.props.initialValue || 0,
  };

  componentDidUpdate = (previousProps, previousState) => {
    const { value } = this.state;

    if (value === previousState.value) return;

    const { name, onChange } = this.props;

    onChange(name, value);
  };

  handleDecrement = () => {
    const value = getControlledInputValue(
      this.props.value,
      undefined,
      this.state.value
    );

    value !== 0 && this.setState({ value: value - 1 });
  };

  handleIncrement = () => {
    const value = getControlledInputValue(
      this.props.value,
      undefined,
      this.state.value
    );

    value !== this.props.max && this.setState({ value: value + 1 });
  };

  render = () => {
    const value = getControlledInputValue(
      this.props.value,
      undefined,
      this.state.value
    );

    return (
      <div className="ui counter">
        <FlexContainer alignItems="center" justifyContent="space-between">
          <Button
            icon={ICON_NAMES.MINUS}
            isOutlined
            isRounded
            onClick={this.handleDecrement}
          />
          <Paragraph>{this.props.renderValue(value)}</Paragraph>
          <Button
            icon={ICON_NAMES.PLUS}
            isOutlined
            isRounded
            onClick={this.handleIncrement}
          />
        </FlexContainer>
      </div>
    );
  };
}

Component.displayName = 'Counter';

Component.defaultProps = {
  initialValue: undefined,
  max: undefined,
  name: '',
  onChange: Function.prototype,
  renderValue: value => value,
  value: undefined,
};

Component.propTypes = {
  /** The default value of the range. Minimum 0. */
  initialValue: PropTypes.number,
  /** The maximum value the input accepts. */
  max: PropTypes.number,
  /** The name for the range. */
  name: PropTypes.string,
  /** A function called when the range value changes. */
  onChange: PropTypes.func,
  /** A function returning a React node for modifying the way values are rendered. */
  renderValue: PropTypes.func,
  /** The current value of the range where it is consumed as a controlled component. Minimum 0. */
  value: PropTypes.number,
};
