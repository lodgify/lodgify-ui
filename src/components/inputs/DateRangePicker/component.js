import moment from 'moment';
import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { uniqueId, debounce } from 'lodash';
import { DateRangePicker } from 'react-dates';

import 'react-dates/initialize';

import { getUpOrDownFromBoolean } from 'utils/get-up-or-down-from-boolean';
import { withResponsive } from 'utils/with-responsive';
import { Icon, ICON_NAMES } from 'elements/Icon';
import { InputController } from 'inputs/InputController';
import { isBlurEvent } from 'utils/is-blur-event';
import { returnFirstArgument } from 'utils/return-first-argument';
import { getWindowHeight } from 'utils/get-window-height';
import { isDisplayedAsModal } from 'utils/is-displayed-as-modal';

import { mapValueToProps } from './utils/mapValueToProps';
import { getNumberOfMonths } from './utils/getNumberOfMonths';
import { MAXIMUM_SCREEN_WIDTH_FOR_TWO_MONTH_CALENDAR } from './constants';

/**
 * A date range picker lets a user pick a date range.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
class Component extends PureComponent {
  state = {
    focusedInput: null,
    windowHeight: null,
  };

  componentDidMount = () => {
    moment.locale(this.props.localeCode);
    this.handleHeightChange();
    global.addEventListener('resize', debounce(this.handleHeightChange, 150));
  };

  componentDidUpdate = (prevProps, prevState) => {
    const { focusedInput: prevFocusedInput } = prevState;
    const { focusedInput } = this.state;
    const { onBlur } = this.props;

    isBlurEvent(prevFocusedInput, focusedInput) && onBlur();
  };

  handleFocusChange = focusedInput => {
    this.setState({ focusedInput });
  };

  handleHeightChange = () => {
    const windowHeight = getWindowHeight();

    if (windowHeight !== this.state.windowHeight) {
      this.setState({
        windowHeight,
      });
    }
  };

  render() {
    const {
      displayFormat,
      endDatePlaceholderText,
      error,
      getIsDayBlocked,
      isValid,
      name,
      onChange,
      startDatePlaceholderText,
      value,
      willOpenAbove,
      windowInnerWidth,
    } = this.props;
    const { focusedInput } = this.state;

    return (
      <InputController
        adaptOnChangeEvent={returnFirstArgument}
        error={error}
        initialValue={null}
        inputOnChangeFunctionName="onDatesChange"
        isFocused={!!focusedInput}
        isValid={isValid}
        mapValueToProps={mapValueToProps}
        name={name}
        onChange={onChange}
        value={value}
      >
        <DateRangePicker
          /* eslint-disable react/jsx-sort-props */
          // Consumer defined props.
          displayFormat={displayFormat}
          endDatePlaceholderText={endDatePlaceholderText}
          isDayBlocked={getIsDayBlocked}
          openDirection={getUpOrDownFromBoolean(willOpenAbove)}
          startDatePlaceholderText={startDatePlaceholderText}
          // Controlled props
          focusedInput={focusedInput}
          // NOTE onDatesChange is required by DateRangePicker but is set in `InputController`
          onDatesChange={Function.prototype}
          onFocusChange={this.handleFocusChange}
          // Static required props.
          endDateId={uniqueId('end_date_id_')}
          startDateId={uniqueId('start_date_id_')}
          // Static custom appearance props.
          customArrowIcon={<Icon name={ICON_NAMES.ARROW_RIGHT} />}
          customInputIcon={<Icon name={ICON_NAMES.CALENDAR} />}
          daySize={52}
          hideKeyboardShortcutsPanel
          navNext={<Icon name={ICON_NAMES.ARROW_RIGHT} />}
          navPrev={<Icon name={ICON_NAMES.ARROW_LEFT} />}
          numberOfMonths={getNumberOfMonths(windowInnerWidth)}
          withPortal={isDisplayedAsModal(this.state.windowHeight)}
          /* eslint-enable react/jsx-sort-props */
        />
      </InputController>
    );
  }
}

Component.displayName = 'DateRangePicker';

Component.defaultProps = {
  displayFormat: 'DD/MM/YYYY',
  endDatePlaceholderText: '',
  error: false,
  getIsDayBlocked: Function.prototype,
  isValid: false,
  localeCode: 'en',
  name: '',
  onBlur: Function.prototype,
  onChange: Function.prototype,
  startDatePlaceholderText: '',
  value: undefined,
  willOpenAbove: false,
  windowInnerWidth: MAXIMUM_SCREEN_WIDTH_FOR_TWO_MONTH_CALENDAR,
};

Component.propTypes = {
  /** The format for inputting and displaying dates as text. */
  displayFormat: PropTypes.oneOfType([PropTypes.string, PropTypes.func]),
  /** The visible placeholder text for the end date input. */
  endDatePlaceholderText: PropTypes.string,
  /** Is the date range picker in an error state. A string is displayed as an error message. */
  error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  /**
   * A function called for each day to be displayed. Returning true blocks that day in the date range picker.
   * @param   {Moment}  day - The day to test.
   * @returns {boolean}     - Is the day blocked.
   */
  getIsDayBlocked: PropTypes.func,
  /** Is the date range picker in a valid state. */
  isValid: PropTypes.bool,
  /** The ISO 639-1 locale code which changes the format and language of days of the week and the months of the year. */
  localeCode: PropTypes.string,
  /** The name for the date range picker. */
  name: PropTypes.string,
  /**
   * Used internally by `Form` so ignored in the styleguide.
   * @ignore
   */
  onBlur: PropTypes.func,
  /**
   * A function called when the date range picker value changes. Dates are wrapped as [Moment objects](https://momentjs.com/docs/#/parsing/).
   * @param {String} name   - `this.props.name`
   * @param {Object} value
   * @param {Moment} value.endDate
   * @param {Moment} value.startDate
   */
  onChange: PropTypes.func,
  /** The visible placeholder text for the start date input. */
  startDatePlaceholderText: PropTypes.string,
  /** The current value of the input where it is consumed as a controlled component. */
  value: PropTypes.shape({
    endDate: PropTypes.object,
    startDate: PropTypes.object,
  }),
  /** Will the calendar open above the input. */
  willOpenAbove: PropTypes.bool,
  /**
   * Is the user on a mobile device.
   * Provided by `withResponsive` so ignored in the styleguide.
   * @ignore
   */
  windowInnerWidth: PropTypes.number,
};

export const ComponentWithResponsive = withResponsive(Component);
