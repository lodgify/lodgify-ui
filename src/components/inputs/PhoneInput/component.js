import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { isEqual } from 'lodash';

import { getIsInputValueReset } from 'utils/get-is-input-value-reset';
import { getControlledInputValue } from 'utils/get-controlled-input-value';

import { InputController } from '../InputController';

import { parseValue } from './utils/parseValue';
import { INITIAL_VALUE } from './constants';
import { getIconOrFlag } from './utils/getIconOrFlag';

/**
 * A phone input allows a user to enter a phone number.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export class Component extends PureComponent {
  state = {
    country: undefined,
    value: INITIAL_VALUE,
  };

  componentDidUpdate(previousProps, previousState) {
    if (getIsInputValueReset(previousProps.value, this.props.value)) {
      this.setState({ country: undefined, value: INITIAL_VALUE });
      return;
    }

    const { value } = this.state;
    const { name, onChange } = this.props;

    !isEqual(previousState.value, value) && onChange(name, value);
  }

  handleChange = (name, value) => {
    const { country, phone } = parseValue(value);

    this.setState({ country, value: phone });
  };

  render() {
    const { autoComplete, error, isValid, label, name, onBlur } = this.props;
    const { country } = this.state;
    const value = getControlledInputValue(
      this.props.value,
      INITIAL_VALUE,
      this.state.value
    );

    return (
      <InputController
        error={error}
        icon={getIconOrFlag(country)}
        isValid={isValid}
        label={label}
        name={name}
        onChange={this.handleChange}
        value={value}
      >
        <input autoComplete={autoComplete} onBlur={onBlur} type="tel" />
      </InputController>
    );
  }
}

Component.displayName = 'PhoneInput';

Component.defaultProps = {
  autoComplete: null,
  error: false,
  isValid: false,
  label: '',
  name: '',
  onBlur: Function.prototype,
  onChange: Function.prototype,
  value: undefined,
};

Component.propTypes = {
  /** Can the phone input be completed automatically by the browser. See [MDN docs for more](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/autocomplete). */
  autoComplete: PropTypes.string,
  /** Is the phone input in an error state. A string is displayed as an error message. */
  error: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  /** Is the phone input in a valid state. */
  isValid: PropTypes.bool,
  /** The visible label for the phone input. */
  label: PropTypes.string,
  /** The name for the phone input. */
  name: PropTypes.string,
  /**
   * Used internally by `Form` so ignored in the styleguide.
   * @ignore
   */
  onBlur: PropTypes.func,
  /**
   * A function called when the phone input value changes
   * @param {String} name
   * @param {String} value
   */
  onChange: PropTypes.func,
  /** The current value of the input where it is consumed as a controlled component. */
  value: PropTypes.string,
};
