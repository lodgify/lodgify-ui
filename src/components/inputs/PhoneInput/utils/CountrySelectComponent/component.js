import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';

import { Dropdown } from 'inputs/Dropdown';

import { getOptionsWithSearch } from '../getOptionsWithSearch';
import { getAllOptions } from '../getAllOptions';

export class Component extends PureComponent {
  handleChange = (name, value) => this.props.onChange(value);

  render = () => {
    const { onFocus, options, value } = this.props;

    return (
      <Dropdown
        getOptionsWithSearch={getOptionsWithSearch}
        isClearable={false}
        isFluid
        onChange={this.handleChange}
        onFocus={onFocus}
        options={getAllOptions(options)}
        value={value}
      />
    );
  };
}

Component.defaultProps = {
  value: undefined,
};

Component.propTypes = {
  onChange: PropTypes.func.isRequired,
  onFocus: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.object).isRequired,
  value: PropTypes.string,
};

Component.displayName = 'CountrySelectComponent';
