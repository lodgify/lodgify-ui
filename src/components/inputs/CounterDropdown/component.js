import React, { useState } from 'react';
import PropTypes from 'prop-types';
import { Popup } from 'semantic-ui-react';
import classnames from 'classnames';

import { some } from 'utils/some';
import { GUESTS } from 'utils/default-strings';
import { Counter } from 'inputs/Counter';
import { Icon, ICON_NAMES } from 'elements/Icon';
import { FlexContainer } from 'layout/FlexContainer';
import { testidFactory } from 'utils/testid';

const TEST_ID_PREFIX = 'counter-dropdown';

const testid = testidFactory(TEST_ID_PREFIX);

/**
 * A counter dropdown allows a user to select a value from a counter input.
 */
// eslint-disable-next-line jsdoc/require-jsdoc
export const Component = ({
  counterName,
  counterValue,
  dropdownLabel,
  maximumCounterValue,
  onChange,
  willOpenAbove,
}) => {
  const [isOpen, setIsOpen] = useState(false);

  return (
    <Popup
      className="counter-dropdown"
      content={
        <FlexContainer flexDirection="row" justifyContent="space-between">
          <Counter
            max={maximumCounterValue}
            name={counterName}
            onChange={onChange}
            value={counterValue}
          />
        </FlexContainer>
      }
      data-testid={testid()}
      on="click"
      onClose={() => setIsOpen(false)}
      onOpen={() => setIsOpen(true)}
      open={isOpen}
      position={willOpenAbove ? 'top left' : 'bottom left'}
      text={dropdownLabel}
      trigger={
        <div
          className={classnames('dropdown-container', {
            dirty: some(counterValue),
          })}
          role="button"
        >
          <div
            className={classnames(
              'ui selection dropdown',
              'counter-dropdown-trigger',
              {
                active: isOpen,
              }
            )}
          >
            <span>{dropdownLabel}</span>
            <Icon name={ICON_NAMES.CARET_DOWN}></Icon>
          </div>
        </div>
      }
    />
  );
};

Component.displayName = 'CounterDropdown';

Component.defaultProps = {
  counterName: undefined,
  counterValue: undefined,
  dropdownLabel: GUESTS,
  maximumCounterValue: undefined,
  onChange: undefined,
  willOpenAbove: false,
};

Component.propTypes = {
  /** The name for the counter. */
  counterName: PropTypes.string,
  /** The value of the counter. Minimum 0. */
  counterValue: PropTypes.number,
  /** The text that populates the dropdown's label. */
  dropdownLabel: PropTypes.string,
  /** The maximum value the counter accepts. */
  maximumCounterValue: PropTypes.number,
  /** A function called when the counter value changes. */
  onChange: PropTypes.func,
  /** Will the dropdown open above or below the trigger */
  willOpenAbove: PropTypes.bool,
};
