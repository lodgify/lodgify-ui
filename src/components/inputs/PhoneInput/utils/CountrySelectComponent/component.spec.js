jest.mock('../getAllOptions');

import React from 'react';
import { shallow, mount } from 'enzyme';
import { expectComponentToHaveDisplayName } from '@lodgify/enzyme-jest-expect-helpers';

import { getAllOptions } from '../getAllOptions';

import { Component as CountrySelectComponent } from './component';

const props = {
  onChange: () => {},
  onFocus: () => {},
  options: [],
};

const getCountrySelectComponent = extraProps =>
  shallow(<CountrySelectComponent {...props} {...extraProps} />);

describe('CountrySelectComponent', () => {
  describe('handleChange', () => {
    it('should call `props.onChange` with the right arguments', () => {
      const onChange = jest.fn();
      const wrapper = getCountrySelectComponent({ onChange });

      const name = 'some name';
      const value = 'some value';

      wrapper.instance().handleChange(name, value);

      expect(onChange).toHaveBeenCalledWith(value);
    });
  });

  describe('render', () => {
    it('should call `getAllOptions` with the right arguments', () => {
      const wrapper = getCountrySelectComponent();

      getAllOptions.mockClear();
      wrapper.instance().render();

      expect(getAllOptions).toHaveBeenCalledWith(props.options);
    });

    it('should render the right structure', () => {
      getAllOptions.mockReturnValueOnce([
        { text: 'some text', value: 'some value ' },
      ]);
      const wrapper = mount(<CountrySelectComponent {...props} />);

      expect(wrapper).toMatchSnapshot();
    });
  });

  it('should have `displayName` `CountrySelectComponent`', () => {
    expectComponentToHaveDisplayName(
      CountrySelectComponent,
      'CountrySelectComponent'
    );
  });
});
