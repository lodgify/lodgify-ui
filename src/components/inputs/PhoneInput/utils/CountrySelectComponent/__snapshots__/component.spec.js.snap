// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CountrySelectComponent render should render the right structure 1`] = `
<CountrySelectComponent
  onChange={[Function]}
  onFocus={[Function]}
  options={Array []}
>
  <Dropdown
    error={false}
    getOptionsWithSearch={[Function]}
    icon={null}
    isClearable={false}
    isCompact={false}
    isDisabled={false}
    isFluid={true}
    isSearchable={false}
    isValid={false}
    label=""
    name=""
    noResultsText="No results"
    onChange={[Function]}
    onFocus={[Function]}
    options={
      Array [
        Object {
          "text": "some text",
          "value": "some value ",
        },
      ]
    }
    willOpenAbove={false}
  >
    <div
      className="dropdown-container dirty"
    >
      <Dropdown
        additionLabel="Add "
        additionPosition="top"
        clearable={false}
        closeOnBlur={true}
        closeOnEscape={true}
        compact={false}
        deburr={false}
        disabled={false}
        icon={
          <Icon
            className={null}
            color={null}
            hasBorder={false}
            isButton={false}
            isCircular={false}
            isColorInverted={false}
            isDisabled={false}
            isLabelLeft={false}
            labelText={null}
            labelWeight={null}
            name="caret down"
            path={null}
            size={null}
          />
        }
        minCharacters={1}
        noResultsMessage="No results"
        onChange={[Function]}
        onClose={[Function]}
        onOpen={[Function]}
        openOnFocus={true}
        options={
          Array [
            Object {
              "key": "0",
              "label": undefined,
              "text": "some text",
              "value": "some value ",
            },
          ]
        }
        placeholder=""
        renderLabel={[Function]}
        search={[Function]}
        searchInput="text"
        selectOnBlur={false}
        selectOnNavigation={true}
        selection={true}
        upward={false}
        wrapSelection={true}
      >
        <Ref
          innerRef={
            Object {
              "current": <div
                aria-disabled="false"
                aria-expanded="false"
                class="ui search selection dropdown"
                role="combobox"
              >
                <input
                  aria-autocomplete="list"
                  autocomplete="off"
                  class="search"
                  tabindex="0"
                  type="text"
                  value=""
                />
                <div
                  aria-atomic="true"
                  aria-live="polite"
                  class="text"
                  role="alert"
                />
                <i
                  class="icon"
                >
                  <svg
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20,8.5l-8,7-8-7Z"
                      fill="currentColor"
                    />
                  </svg>
                </i>
                <div
                  class="menu transition"
                  role="listbox"
                >
                  <div
                    aria-checked="false"
                    aria-selected="true"
                    class="selected item"
                    role="option"
                    style="pointer-events: all;"
                  >
                    <span
                      class="text"
                    >
                      some text
                    </span>
                  </div>
                </div>
              </div>,
            }
          }
        >
          <RefFindNode
            innerRef={
              Object {
                "current": <div
                  aria-disabled="false"
                  aria-expanded="false"
                  class="ui search selection dropdown"
                  role="combobox"
                >
                  <input
                    aria-autocomplete="list"
                    autocomplete="off"
                    class="search"
                    tabindex="0"
                    type="text"
                    value=""
                  />
                  <div
                    aria-atomic="true"
                    aria-live="polite"
                    class="text"
                    role="alert"
                  />
                  <i
                    class="icon"
                  >
                    <svg
                      viewBox="0 0 24 24"
                    >
                      <path
                        d="M20,8.5l-8,7-8-7Z"
                        fill="currentColor"
                      />
                    </svg>
                  </i>
                  <div
                    class="menu transition"
                    role="listbox"
                  >
                    <div
                      aria-checked="false"
                      aria-selected="true"
                      class="selected item"
                      role="option"
                      style="pointer-events: all;"
                    >
                      <span
                        class="text"
                      >
                        some text
                      </span>
                    </div>
                  </div>
                </div>,
              }
            }
          >
            <div
              aria-disabled={false}
              aria-expanded={false}
              className="ui search selection dropdown"
              onBlur={[Function]}
              onChange={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onMouseDown={[Function]}
              role="combobox"
            >
              <Ref
                innerRef={
                  Object {
                    "current": <input
                      aria-autocomplete="list"
                      autocomplete="off"
                      class="search"
                      tabindex="0"
                      type="text"
                      value=""
                    />,
                  }
                }
              >
                <RefFindNode
                  innerRef={
                    Object {
                      "current": <input
                        aria-autocomplete="list"
                        autocomplete="off"
                        class="search"
                        tabindex="0"
                        type="text"
                        value=""
                      />,
                    }
                  }
                >
                  <DropdownSearchInput
                    autoComplete="off"
                    key="text"
                    onChange={[Function]}
                    style={
                      Object {
                        "width": undefined,
                      }
                    }
                    tabIndex={0}
                    type="text"
                    value=""
                  >
                    <input
                      aria-autocomplete="list"
                      autoComplete="off"
                      className="search"
                      onChange={[Function]}
                      style={
                        Object {
                          "width": undefined,
                        }
                      }
                      tabIndex={0}
                      type="text"
                      value=""
                    />
                  </DropdownSearchInput>
                </RefFindNode>
              </Ref>
              <div
                aria-atomic={true}
                aria-live="polite"
                className="text"
                role="alert"
              />
              <Icon
                className=""
                color={null}
                hasBorder={false}
                isButton={false}
                isCircular={false}
                isColorInverted={false}
                isDisabled={false}
                isLabelLeft={false}
                labelText={null}
                labelWeight={null}
                name="caret down"
                onClick={[Function]}
                path={null}
                size={null}
              >
                <i
                  className="icon"
                  onClick={[Function]}
                >
                  <svg
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M20,8.5l-8,7-8-7Z"
                      fill="currentColor"
                    />
                  </svg>
                </i>
              </Icon>
              <DropdownMenu
                role="listbox"
              >
                <div
                  className="menu transition"
                  role="listbox"
                >
                  <DropdownItem
                    active={false}
                    key="0"
                    onClick={[Function]}
                    selected={true}
                    style={
                      Object {
                        "pointerEvents": "all",
                      }
                    }
                    text="some text"
                    value="some value "
                  >
                    <div
                      aria-checked={false}
                      aria-selected={true}
                      className="selected item"
                      onClick={[Function]}
                      role="option"
                      style={
                        Object {
                          "pointerEvents": "all",
                        }
                      }
                    >
                      <span
                        className="text"
                      >
                        some text
                      </span>
                    </div>
                  </DropdownItem>
                </div>
              </DropdownMenu>
            </div>
          </RefFindNode>
        </Ref>
      </Dropdown>
    </div>
  </Dropdown>
</CountrySelectComponent>
`;
